python/cuml/dask/ensemble/base.py

```diff
@@ -0,0 +1,179 @@
+import dask
+import math
+
+from cuml.dask.common.input_utils import DistributedDataHandler, \
+    wait_and_raise_from_futures, concatenate
+from dask.distributed import default_client
+
+
+class BaseRandomForestModel(object):
+    """
+    BaseRandomForestModel defines functions used in both Random Forest
+    Classifier and Regressor for Multi Node and Multi GPU models. The common
+    functions are defined here and called from the main Random Forest Multi
+    Node Multi GPU APIs. The functions defined here are not meant to be used
+    as a part of the public API.
+    """
+
+    def _create_model(self, model_func,
+                      client,
+                      workers,
+                      n_estimators,
+                      base_seed,
+                      **kwargs):
+
+        self.client = default_client() if client is None else client
+        if workers is None:
+            workers = self.client.has_what().keys()

```

Comment:
I recently found a more performant way of doing this, which doesn't require fetching the dictionary of all the partitions on all the workers: `self.client.scheduler_info()['workers'].keys()`.